daemon/containerd/image_push.go

```diff
@@ -45,7 +45,13 @@ func (i *ImageService) PushImage(ctx context.Context, targetRef reference.Named,
 	if err != nil {
 		return err
 	}
-	defer release(leasedCtx)
+	defer func() {
+		err := release(leasedCtx)
+		if err != nil && !cerrdefs.IsNotFound(err) {
+			logrus.WithField("image", targetRef).WithError(err).Error("failed to delete lease created for push")
```

Comment:
I guess this could be a `Warn()` instead of an `Error()` (given that we're able to proceed, and there's nothing immediately actionable)